### PR TITLE
Sync: re-implement `grade-school` according to new specs

### PR DESCRIPTION
**Pull request by jiegillet**
This one is the biggest one of the batch.
Some of the specs changed, and the tests insisted on knowing if adding a student succeeded or not, so I re-implemented the example solution and all of the tests. Please review carefully.

All changes:
* synced instructions and metadata (automatic)
* re-implemented all 20 tests requested by configlet
* synced `tests.toml` (automatic)
* updated example solution and stub
* added my name to `config.json`

### CODE REVIEW

File: exercises/practice/grade-school/.meta/src/GradeSchool.example.elm [hunk 1]
```diff
@@ -35,6 +44,8 @@ studentsInGrade grade school =
             []
 
 
-allStudents : School -> List ( Grade, List Student )
+allStudents : School -> List Student
```
Comment (ceddlyburge) [inline exercises/practice/grade-school/.meta/src/GradeSchool.example.elm:47]: Does this rely on the fact that Dict.toList returns a list sorted by the key? If so probably worth a comment saying so ...

File: exercises/practice/grade-school/src/GradeSchool.elm [hunk 2]
```diff
@@ -12,15 +12,20 @@ type alias Student =
...
-    Dict.empty
+type Result
+    = Added
+    | Duplicate
 
 
-addStudent : Grade -> Student -> School -> School
+emptySchool : School
+emptySchool =
+    Debug.todo "Please implement this function"
```
[the remark this answers is not in the document]
Comment (jiegillet) [reply exercises/practice/grade-school/src/GradeSchool.elm]: I did it, but isn't that message standard?

File: exercises/practice/grade-school/tests/Tests.elm [hunk 3]
```diff
@@ -1,67 +1,166 @@
...
-                            |> addStudent 1 "Charlie"
-                            |> allStudents
-                        )
-        , skip <|
-            test "get students in a non-existent grade" <|
-                \() -> Expect.equal [] (studentsInGrade 1 GradeSchool.empty)
+        [ test "Roster is empty when no student is added" <|
+            \() -> Expect.equal [] (allStudents emptySchool)
+        , test "Add a student" <|
+            \() ->
```
[the remark this answers is not in the document]
Comment (jiegillet) [reply exercises/practice/grade-school/tests/Tests.elm]: Done.
I also realized I forgot the `skip`.

File: exercises/practice/grade-school/.meta/src/GradeSchool.example.elm (hunk 1 above)
Comment (jiegillet) [reply exercises/practice/grade-school/.meta/src/GradeSchool.example.elm:47]: Yes it does. It's clear on the documentation, so it should be reliable.
I added a comment.

File: exercises/practice/grade-school/src/GradeSchool.elm (hunk 2 above)
[the remark this answers is not in the document]
Comment (ceddlyburge) [reply exercises/practice/grade-school/src/GradeSchool.elm]: Well it is common, but Erik asked me to do it on an exercise I did recently, and it does make sense, especially when there are multiple functions to implement ...

**Review by ceddlyburge — APPROVED** [review]

**Pull request merged**